### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by rickstylz01**

### CODE REVIEW

File: src/controllers/authControllers.js [hunk 1]
```diff
@@ -64,51 +94,178 @@ exports.signup = async (req, res) => {
...
+      token: uuidv4(),
+      expiresIn: moment().add(1, "hours"),
+      tokenType: "password-reset",
+    });
 
-    // Get the token from the authorization header by removing "Bearer " from the string.
-    let token = req.headers.authorization.replace("Bearer ", "");
+    const to = user.email;
+    const subject = "Fogot your password?";
+    const html = `<p>Click <a href="reset-form-with-token>here</a> to reset your password.</p>`;
```
Comment (TheycallmeNate) [inline src/controllers/authControllers.js:209]: The href does not link to anywhere but let see what design does before we make any changes to this flow.

File: src/models/userModel.js [hunk 2]
```diff
@@ -4,11 +4,11 @@ const UserSchema = new mongoose.Schema(
   {
```
Comment (TheycallmeNate) [inline src/models/userModel.js:4]: Can you confirm with Hope about the max characters for all the properties that have that property?

**Review by TheycallmeNate — COMMENTED** [review]
The biggest thing that needs to be done is sending the verification token in the email from the signup function in.

File: src/controllers/authControllers.js [hunk 3]
```diff
@@ -64,51 +94,178 @@ exports.signup = async (req, res) => {
   }
 };
 
-//@route POST '/signup'
-//@desc Register/Signup a new user
-//@access Public
+exports.emailVerification = async (req, res) => {
+  try {
+    let token = await TokenModel.findOne({ token: req.params.token });
```
[the remark this answers is not in the document]
Comment (rickstylz01) [reply src/controllers/authControllers.js:99]: I see what you mean thanks.  I will fix this and then make a new branch when I start setting up nodemailer.

**Review by TheycallmeNate — APPROVED** [review]

**Pull request merged**